Spigot-Server-Patches/0325-World-EntityHuman-Lookup-Optimizations.patch

```diff
@@ -0,0 +1,102 @@
+                 EntityHuman entityhuman = (EntityHuman) entity;
+ 
+                 this.players.add(entityhuman);
++                this.playersByName.put(entityhuman.getName(), entityhuman);
++                // Paper end
+                 this.everyoneSleeping();
+             }
+ 
+@@ -1101,6 +1104,7 @@ public abstract class World implements GeneratorAccess, IIBlockAccess, AutoClose
+         entity.die();
+         if (entity instanceof EntityHuman) {
+             this.players.remove(entity);
++            this.playersByName.remove(entity.getName()); // Paper - World EntityHuman Lookup Optimizations
+             // Spigot start
+             for ( Object o : worldMaps.d )
+             {
+@@ -1131,6 +1135,7 @@ public abstract class World implements GeneratorAccess, IIBlockAccess, AutoClose
+         entity.die();
+         if (entity instanceof EntityHuman) {
+             this.players.remove(entity);
++            this.playersByName.remove(entity.getName()); // Paper - World EntityHuman Lookup Optimizations
+             this.everyoneSleeping();
+         }
+ 
+@@ -2659,6 +2664,8 @@ public abstract class World implements GeneratorAccess, IIBlockAccess, AutoClose
+ 
+     @Nullable
+     public EntityHuman a(String s) {
++        // Paper start - World EntityHuman Lookup Optimizations
++        /*
+         for (int i = 0; i < this.players.size(); ++i) {
+             EntityHuman entityhuman = (EntityHuman) this.players.get(i);
+ 
+@@ -2668,10 +2675,15 @@ public abstract class World implements GeneratorAccess, IIBlockAccess, AutoClose
+         }
+ 
+         return null;
++        */
++        return this.playersByName.get(s);
++        // Paper end
+     }
+ 
+     @Nullable
+     public EntityHuman b(UUID uuid) {
++        // Paper start - World EntityHuman Lookup Optimizations
++        /*
+         for (int i = 0; i < this.players.size(); ++i) {
+             EntityHuman entityhuman = (EntityHuman) this.players.get(i);
+ 
+@@ -2681,6 +2693,10 @@ public abstract class World implements GeneratorAccess, IIBlockAccess, AutoClose
+         }
+ 
+         return null;
++        */
++        Entity entity = ((WorldServer)this).entitiesByUUID.get(uuid);
++        return entity instanceof EntityHuman ? (EntityHuman) entity : null;
++        // Paper end
+     }
+ 
+     public void checkSession() throws ExceptionWorldConflict {
+diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
+index b57e1ff36..60798953a 100644
+--- a/src/main/java/net/minecraft/server/WorldServer.java
++++ b/src/main/java/net/minecraft/server/WorldServer.java
+@@ -1029,6 +1029,7 @@ public class WorldServer extends World implements IAsyncTaskHandler {
+         }
+ 
+         Entity old = this.entitiesByUUID.put(entity.getUniqueID(), entity);
++        if (entity instanceof EntityHuman) this.playersByName.put(entity.getName(), (EntityHuman) entity); // Paper - World EntityHuman Lookup Optimizations
```

Comment:
Since you manage them in World.java where the list version is maintained, these actually aren't needed.